src/components/Voting/Referendum/Items/ProposalItem/index.js

```diff
@@ -100,7 +104,7 @@ function ProposalItem({
             <div>
               <div className={styles.metaTextInfo}>
                 Proposed by:
-                <b>{ proposer && truncate(proposer, 13) }</b>
+                <b>{ findNameOrId(proposer, usersList) }</b>
```

Comment:
How about instead of searching through the list for every entry, we preprocess it into a map and do direct accesses?